### PR TITLE
chore: add no-vars rule to match sonar cloud

### PR DESCRIPTION
**Pull request by chmelevskij**
match some of the rules to sonar cloud to reduce the amount on PRs

### CODE REVIEW

**Comment by haslinghuis** [comment]
Yeah  src/js/protocols/stm32usbdfu.js was the only file I didn't touch yet.

**Review by haslinghuis — APPROVED** [review]

**Review by ctzsnooze — DISMISSED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by haslinghuis — APPROVED** [review]

**Review by limonspb — APPROVED** [review]

**Review by Asizon — APPROVED** [review]

**Pull request merged**
